server/src/main/java/org/opensearch/index/mapper/ObjectMapper.java

```diff
@@ -291,6 +291,15 @@ protected static boolean parseObjectOrDocumentTypeProperties(
             } else if (fieldName.equals("enabled")) {
                 builder.enabled(XContentMapValues.nodeBooleanValue(fieldNode, fieldName + ".enabled"));
                 return true;
+            } else if (fieldName.equals("derived")) {
+                if (fieldNode instanceof Collection && ((Collection) fieldNode).isEmpty()) {
+                    // nothing to do here, empty (to support "derived: []" case)
+                } else if (!(fieldNode instanceof Map)) {
+                    throw new OpenSearchParseException("derived must be a map type");
+                } else {
+                    parseDerived(builder, (Map<String, Object>) fieldNode, parserContext);
+                }
```

Comment:
Can you reorder the logic here? The `if (!foo) { // doA } else { // doB }` logic is harder to read than `if (foo) { // doB } else { // doA }`

Instead of `else if (!(fieldNode instanceof Map))`, could you do:
```
else if (fieldNode instanceof Map) {
  parseDerived(...);
} else {
  throw new OpenSearchParseException(...);
}
```

Reply:
Makes sense